File: java/gpt/p10_nre.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class p10_nre {
    private static int getValueFromArray(int[] arr, int idx) {
        if (arr == null || idx < 0 || idx >= arr.length) {
            return -1;
        }
        return arr[idx];
    }

    public static void main(String[] args) throws IOException {
        int[] idSequence = {10, 20, 30, 40, 50};

        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        String line = reader.readLine();
        if (line == null) {
            System.exit(-1);
            return;
        }

        int idx;
        try {
            idx = Integer.parseInt(line.trim());
        } catch (NumberFormatException ex) {
            System.exit(-1);
            return;
        }

        int result = getValueFromArray(idSequence, idx);
        System.exit(result);
    }
}

File: python/gpt/p10_nre.py
import sys
from typing import List


def getValueFromArray(arr: List[int], idx: int) -> int:
    if arr is None or idx < 0 or idx >= len(arr):
        return -1
    return arr[idx]


def main() -> None:
    id_sequence = [10, 20, 30, 40, 50]
    try:
        raw = sys.stdin.readline()
    except Exception:
        sys.exit(-1)
        return

    if not raw:
        sys.exit(-1)
        return

    try:
        idx = int(raw.strip())
    except ValueError:
        sys.exit(-1)
        return

    result = getValueFromArray(id_sequence, idx)
    sys.exit(result)


if __name__ == "__main__":
    main()
